src/utils.ts

```diff
@@ -593,11 +592,29 @@ export const backportImpl = async (
           labelsToAdd.push(BACKPORT_REQUESTED_LABEL);
         }
 
-        const semverLabel = pr.labels.find((l: any) =>
-          l.name.startsWith(SEMVER_PREFIX),
-        );
+        const semverLabel = labelUtils.getSemverLabel(pr);
         if (semverLabel) {
-          labelsToAdd.push(semverLabel.name);
+          // If the new PR for some reason has a semver label already, then
+          // we need to compare the two semver labels and ensure the higher one
+          // takes precedence.
+          const newPRSemverLabel = labelUtils.getSemverLabel(newPr);
+          if (newPRSemverLabel) {
+            const higherLabel = labelUtils.getHighestSemverLabel(
+              semverLabel.name,
+              newPRSemverLabel.name,
+            );
+            // The existing label is lower precedence - remove and replace it.
+            if (higherLabel === semverLabel.name) {
```

Comment:
I might be misreading this but I think this should be `!==`?

```suggestion
            if (higherLabel !== semverLabel.name) {
```

Reply:
this says "if the higher label is the one on the original PR and not the one currently on the PR, replace it"